modules/file-extension-common/src/main/java/org/mule/runtime/module/extension/file/api/FileSystem.java

```diff
@@ -219,6 +220,8 @@ void write(String filePath,
      */
     PathLock lock(Path path, Object... params);
 
+    Lock createMuleLock(String id);
```

Comment:
Why would this class expose a createMuleLock method?. Makes sense to use it internally but it should not be exposed to end users.